File: frontend/types/issue.ts
export type IssueStatus = "open" | "in_progress" | "closed";
export type IssuePriority = "low" | "medium" | "high";

export interface Attachment {
  id: number;
  issue_id: number;
  file_name: string;
  file_url: string;
  public_id?: string;
  resource_type: string;
  file_type?: string;
  file_size?: number;
  created_at: string;
}

export interface Issue {
  id: number;
  title: string;
  description?: string;
  status: IssueStatus;
  priority: IssuePriority;
  reporter?: string;
  assignee?: string;
  created_at: string;
  updated_at?: string;
  attachments?: Attachment[];
}

// shape needed when creating a new issue (no id/timestamps yet)
export interface IssueCreate {
  title: string;
  description?: string;
  status?: IssueStatus;
  priority?: IssuePriority;
  reporter?: string;
  assignee?: string;
}

File: backend/app/models/issue.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.db.base_class import Base


class Issue(Base):
    __tablename__ = "issues"

    id          = Column(Integer, primary_key=True, index=True)
    title       = Column(String, nullable=False)
    description = Column(String, nullable=True)
    status      = Column(String, default="open")        # open / in_progress / closed
    priority    = Column(String, default="medium")      # low / medium / high
    organization_id = Column(Integer, ForeignKey("organizations.id"), nullable=True)

    # Legacy free-text fields (kept for backward compatibility)
    reporter    = Column(String, nullable=True)
    assignee    = Column(String, nullable=True)

    # FK relationships to User (added in parallel — non-breaking)
    reporter_id = Column(Integer, ForeignKey("users.id"), nullable=True)
    assignee_id = Column(Integer, ForeignKey("users.id"), nullable=True)

    created_at  = Column(DateTime(timezone=True), server_default=func.now())
    updated_at  = Column(DateTime(timezone=True), onupdate=func.now())

    # Relationships
    organization= relationship("Organization", back_populates="issues")

    attachments = relationship(
        "IssueAttachment",
        back_populates="issue",
        cascade="all, delete-orphan",
    )
    reporter_user = relationship(
        "User",
        foreign_keys=[reporter_id],
        back_populates="reported_issues",
    )
    assignee_user = relationship(
        "User",
        foreign_keys=[assignee_id],
        back_populates="assigned_issues",
    )


class IssueAttachment(Base):
    __tablename__ = "issue_attachments"

    id            = Column(Integer, primary_key=True, index=True)
    issue_id      = Column(Integer, ForeignKey("issues.id"), nullable=False)
    file_name     = Column(String, nullable=False)
    file_url      = Column(String, nullable=False)
    public_id     = Column(String, nullable=True)
    resource_type = Column(String, nullable=False)
    file_type     = Column(String, nullable=True)
    file_size     = Column(Integer, nullable=True)
    created_at    = Column(DateTime(timezone=True), server_default=func.now())

    issue = relationship("Issue", back_populates="attachments")
